packages/ui-test-utils/src/utils/generateComponentExamples.ts

```diff
@@ -28,16 +28,54 @@ import { generatePropCombinations } from './generatePropCombinations'
 import React, { ComponentType, ReactNode } from 'react'
 
 export type StoryConfig<Props> = {
-  sectionProp?: keyof Props // TODO make this mandatory
+  /**
+   * Used to divide the resulting examples into sections. It should correspond
+   * to an enumerated prop in the Component
+   */
+  sectionProp?: keyof Props
+  /**
+   * Specifies the max number of examples that can exist in a single page
+   * within a section
+   */
   maxExamplesPerPage?: number | ((sectionName: string) => number)
+  /**
+   * Specifies the total max number of examples. Default: 500
+   */
   maxExamples?: number
-  propValues?: Partial<Record<keyof Props, any[]>>
+  /**
+   * An object with keys that correspond to the component props. Each key has a
+   * corresponding value array. This array contains possible values for that prop.
+   */
+  propValues?: Partial<Record<keyof Props | string, any[]>>
```

Comment:
I added to these types `| string` to make it possible to add custom fields. No code modification was needed.